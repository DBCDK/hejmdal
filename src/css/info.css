.header {
  font-size: 1.4em;
}

.title, .body {
  text-align: left;
}

.close, .title {
  font-size: 1.5em;
}

.body {
  font-size: 1.1em;
}
