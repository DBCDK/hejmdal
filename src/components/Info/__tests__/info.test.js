import sinon from 'sinon';
import {showInfo} from '../info.component';
import {mockContext} from '../../../utils/test.util';

describe('test showinfo method', () => {
  let ctx;
  let sandbox;

  beforeEach(() => {
    ctx = mockContext();
    sandbox = sinon.sandbox.create();
  });

  afterEach(() => {
    sandbox.restore();
  });

  it('Should render cookie info page', async () => {
    const spy = sandbox.spy(ctx, 'render');
    ctx.params = {
      infoId: 'cookies'
    };
    showInfo(ctx, ctx);
    expect(spy.called).toBe(true);
    expect(typeof spy.args[0][1].textObj).toBe('object');
  });

  it('Should not render info page', async () => {
    const spy = sandbox.spy(ctx, 'render');
    const next = sandbox.mock();
    ctx.params = {
      infoId: 'not_valid'
    };
    showInfo(ctx, ctx, next);
    expect(next.called).toBe(true);
    expect(spy.called).toBe(false);
  });
});
