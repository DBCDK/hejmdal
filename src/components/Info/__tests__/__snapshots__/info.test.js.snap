// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test showinfo method Should render cookie info page 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Info",
      Object {
        "info": true,
        "textObj": Object {
          "sections": Array [
            Object {
              "texts": Array [
                "Vi benytter cookies til at gemme information om dine logins.",
              ],
            },
            Object {
              "header": "Valgte biblioteker",
              "texts": Array [
                "Vi gemmer information om, hvilke biblioteker, der har været brugt ved login på denne computer, telefon, tablet osv. Det gør vi for at kunne tilbyde dig en liste over tidligere benyttede biblioteker, når du efterfølgende logger ind. Dermed kan vi give dig mulighed for et hurtigere login.",
              ],
            },
            Object {
              "header": "Login",
              "texts": Array [
                "Vi gemmer information om, at du er logget ind og hvilket bibliotek, du er logget ind ved, således at du ikke skal logge ind mere end én gang. Denne information fjernes, når du enten logger ud eller lukker browseren.",
              ],
            },
            Object {
              "header": "Slå cookies fra i browseren ",
              "texts": Array [
                "Bibliotekslogin virker ikke uden cookies. Du kan have ændret indstillingerne i din browser, så du ikke accepterer cookies. I så fald vil du få en fejlside, når du forsøger at benytte Bibliotekslogin.",
              ],
            },
            Object {
              "texts": Array [
                "Alle browsere tillader, at du sletter dine cookies. Hvis du gør det, mens du er logget ind, kan du derfor også opleve fejl på siden.",
              ],
            },
            Object {
              "header": "Cookies hos andre hjemmesider",
              "texts": Array [
                "Bibliotekslogin benytter ikke andre tjenester, der anvender cookies hos brugeren.",
              ],
            },
          ],
          "title": "Cookies, formål og relevans",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
